### PR TITLE
Closes #2624: Open Trellis events in new tab

### LINKED ISSUE #2624: Add context aware behavior for opening external links in a new tab.
### Is your feature request related to a problem? Please describe.
When looking at a calendar of events, users might be browsing events rather than looking for a specific event. In the current iteration, if someone clicks on a Trellis Event they are taken to a new website kind of unexpectedly and lose the context of the other events on the calendar.

This is a request from Pila as a requirement for the University Calendar.

### Describe the solution you'd like
If an Event is going to take you to a new website (like an imported Trellis Event), then it should open in a new tab, so that people don't lose their place on the calendar.

### Describe alternatives you've considered
An alternative approach would be to stay on the site and display the Event node of the imported event, with an option to click out to events with additional Registration or Session information.

### PR DESCRIPTION
**Pull request by danahertzberg**
- [ ] Is Quickstart link an issue?

## Description

Define 'target="_blank"' for Trellis events links

## Related issues

Closes #2624

## How to test

Example Trellis event id: a3U3n00000IqSFPEA3

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [x] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_event/az_event_trellis/az_event_trellis.module [hunk 1]
```diff
@@ -79,3 +79,12 @@ function az_event_trellis_migration_plugins_alter(array &$migrations) {
   }
 
 }
+
+/**
+ * Implements hook_link_alter().
+ */
+function az_event_trellis_link_alter(&$variables) {
+  if ($variables['url']->isExternal()) {
```
Comment (trackleft) [inline modules/custom/az_event/az_event_trellis/az_event_trellis.module:87]: Is `$variables['url']` an actual object?

**Comment by danahertzberg** [comment]
Priority for 2.8.x release

**Comment by danahertzberg** [comment]
Next steps: Add contrib module and see issue for desired defaults.

**Comment by danahertzberg** [comment]
`extlink` module on Drupal https://www.drupal.org/project/extlink

**Comment by danahertzberg** [comment]
Closing in favor of https://github.com/az-digital/az_quickstart/pull/2762

**Pull request closed without merging**